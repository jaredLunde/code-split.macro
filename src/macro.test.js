import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-plugin-macros'

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {filename: __filename},
  tests: [
    `
      import codeSplit from './macro'
      const Home = codeSplit('./macro', true)
    `,
    `
      import codeSplit from './macro'
      const Home = codeSplit('./macro', false)
    `,
    `
      import codeSplit from './macro'
      const Home = codeSplit('./macro', false)
      const OtherHome = codeSplit('./macro', false)
    `,
  ],
})
