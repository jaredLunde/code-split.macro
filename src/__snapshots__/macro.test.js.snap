// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `

import codeSplit from './macro'
const Home = codeSplit('./macro', true)

      ↓ ↓ ↓ ↓ ↓ ↓

const Home = () => require('/Users/jaredlunde/oss/code-split.macro/src/macro')


`;

exports[`macros 2. macros: 2. macros 1`] = `

import codeSplit from './macro'
const Home = codeSplit('./macro', false)

      ↓ ↓ ↓ ↓ ↓ ↓

const Home = () =>
  import(
    /* webpackChunkName: "code-split.macro/src/macro" */
    '/Users/jaredlunde/oss/code-split.macro/src/macro'
  )


`;

exports[`macros 3. macros: 3. macros 1`] = `

import codeSplit from './macro'
const Home = codeSplit('./macro', false)
const OtherHome = codeSplit('./macro', false)

      ↓ ↓ ↓ ↓ ↓ ↓

const Home = () =>
  import(
    /* webpackChunkName: "code-split.macro/src/macro" */
    '/Users/jaredlunde/oss/code-split.macro/src/macro'
  )

const OtherHome = () =>
  import(
    /* webpackChunkName: "code-split.macro/src/macro" */
    '/Users/jaredlunde/oss/code-split.macro/src/macro'
  )


`;

exports[`macros 4. macros: 4. macros 1`] = `

import codeSplit from './macro'
const Home = codeSplit('./macro', __SERVER__)

      ↓ ↓ ↓ ↓ ↓ ↓

const Home = __SERVER__
  ? () => require('/Users/jaredlunde/oss/code-split.macro/src/macro')
  : () =>
      import(
        /* webpackChunkName: "code-split.macro/src/macro" */
        '/Users/jaredlunde/oss/code-split.macro/src/macro'
      )


`;
